src/BlockTypes/ProductSaleBadge.php

```diff
@@ -63,6 +63,68 @@ protected function get_block_type_supports() {
 		);
 	}
 
+	/**
+	 * Initialize this block.
+	 */
+	protected function initialize() {
+		parent::initialize();
+		add_filter(
+			'wp_theme_json_data_theme',
+			array( $this, 'add_default_sale_badge_styles' )
+		);
+	}
+
+	/**
+	 * Add default styles for Sale Badge block.
+	 *
+	 * @param WP_Theme_JSON_Data $theme_data The theme data object that is passed as a parameter.
+	 * @return array
+	 */
+	public function add_default_sale_badge_styles( $theme_data ) {
+		// Get the existing Sale Badge block styles.
+		$block_styles = isset( $theme_data->get_data()['styles']['blocks']['woocommerce/product-sale-badge'] )
+		? $theme_data->get_data()['styles']['blocks']['woocommerce/product-sale-badge']
+		: array();
```

Comment:
Instead of repeatedly accessing `$theme_data->get_data()`, I would recommend storing its value in a variable so the function can be more efficient. It is also possible to simplify some of the `isset` expressions with null coalescing operators. For example:

```suggestion
		$theme_data_array = $theme_data->get_data();

		// Get the existing Sale Badge block styles.
		$block_styles = $theme_data_array['styles']['blocks']['woocommerce/product-sale-badge'] ?? array();
```

Reply:
That makes sense. Thanks again for your helpful review! 🙌🏻